File: java/ankit_kanyal/devnest_problems/MinimumInRotatedSortedArray.java
// link- https://devsnest.in/algo-challenges/minimum-in-rotated-sorted-array

static int solve(int n, int[] arr){
//CODE HERE 
int firstElement = arr[0], left=0, right=arr.length-1;
int ans=arr[0];
    while(left<right)
    {
        int middle = left+(right-left)/2;
        if(arr[middle]<firstElement){
            right=middle-1;
            ans=arr[middle];
        }
        else{
            left=middle+1;
        }
             
    }

    if(arr[left]<=ans) return arr[left];
    else return ans;
}

File: Python/chandu/Minimum_In_Rotated_Sorted_Array.py
def solve(n, nums):
    # CODE HERE
    if n==1:
        return nums[0]
    
    #two pointers
    l, r = 0, n-1
    if nums[0]<nums[r]:
        return nums[0]
    
    while l<=r:
        mid = (l+r)//2

        if nums[mid]>nums[mid+1]:
            return nums[mid+1]
        if nums[mid-1]>nums[mid]:
            return nums[mid]
        if nums[0]< nums[mid]:
            l = mid+1
        else:
            r = mid-1
